File: Algorithms/1200. Minimum Absolute Difference/swift/solution.sort-1traversals.swift
class Solution {
    func minimumAbsDifference(_ arr: [Int]) -> [[Int]] {
        let sortedArr = arr.sorted()
        var minPairDiff = Int.max
        var answer: [[Int]] = []
        
        for i in 0..<sortedArr.count - 1 {
            let currPairDiff = sortedArr[i + 1] - sortedArr[i]
            
            if currPairDiff == minPairDiff {
                answer.append([sortedArr[i], sortedArr[i + 1]])
            } else if currPairDiff < minPairDiff {
                answer = [[sortedArr[i], sortedArr[i + 1]]]
                minPairDiff = currPairDiff
            }
        }
        
        return answer
    }
}

File: Algorithms/1200. Minimum Absolute Difference/rust/solution.sort-1traversals.rs
impl Solution {
    pub fn minimum_abs_difference(mut arr: Vec<i32>) -> Vec<Vec<i32>> {
        arr.sort_unstable();
        let mut min_pair_diff = i32::MAX;
        let mut answer = Vec::new();
        
        for i in 0..arr.len()-1 {
            let curr_pair_diff = arr[i+1] - arr[i];
            if curr_pair_diff == min_pair_diff {
                answer.push(vec![arr[i], arr[i+1]]);
            } else if curr_pair_diff < min_pair_diff {
                answer = vec![vec![arr[i], arr[i+1]]];
                min_pair_diff = curr_pair_diff;
            }
        }
        
        answer
    }
}
